### PR TITLE
Use InsertSPIRVOp to create calls to barrier SPIR-V intrinsics

### PR DESCRIPTION
**Pull request by kpet**
Add missing test for coherent decorations.

Contributes to #308.

Signed-off-by: Kévin Petit <kpet@free.fr>

### CODE REVIEW

File: lib/AllocateDescriptorsPass.cpp [hunk 1]
```diff
@@ -924,17 +924,16 @@ bool AllocateDescriptorsPass::CallTreeContainsGlobalBarrier(Function *F) {
...
           // barrier()
-          if (auto *semantics = dyn_cast<ConstantInt>(call->getOperand(2))) {
+          if (auto *semantics = dyn_cast<ConstantInt>(call->getOperand(3))) {
             uses_barrier =
                 semantics->getZExtValue() & kMemorySemanticsUniformMemory;
           }
-        } else if (call->getCalledFunction()->getName().equals(
-                       "__spirv_memory_barrier")) {
+        } else if (call->getCalledFunction()->getName().startswith(
+                       "spirv.op.225")) {
```
Comment (dneto0) [inline lib/AllocateDescriptorsPass.cpp]: Need to guard against the case where an instruction number is 2250, for example.
I think this should have a "." after the 225, so: "spirv.op.225."

File: lib/AllocateDescriptorsPass.cpp [hunk 2]
```diff
@@ -924,17 +924,16 @@ bool AllocateDescriptorsPass::CallTreeContainsGlobalBarrier(Function *F) {
         // and StorageBuffer storage classes) semantics are checked because
         // Workgroup variables are inherently coherent (and do not require the
         // decoration).
-        if (call->getCalledFunction()->getName().equals(
-                "__spirv_control_barrier")) {
+        if (call->getCalledFunction()->getName().startswith("spirv.op.224")) {
```
Comment (dneto0) [inline lib/AllocateDescriptorsPass.cpp]: Should add a period after the 224.  Guards against an instruction numbered 2240, for example.

File: test/CommonBuiltins/no_duplicate_barrier.ll [hunk 3]
```diff
@@ -2,7 +2,7 @@
 ; RUN: FileCheck %s < %t
 
 ; CHECK: noduplicate
-; CHECK-NEXT: __spirv_control_barrier
+; CHECK-NEXT: spirv.op.224
```
Comment (dneto0) [inline test/CommonBuiltins/no_duplicate_barrier.ll]: add a period at the end.

File: lib/AllocateDescriptorsPass.cpp (hunk 1 above)
Comment (kpet) [reply lib/AllocateDescriptorsPass.cpp]: Yes, better safe than sorry.

**Pull request merged**